core/src/main/java/io/lionweb/utils/ValidationResult.java

```diff
@@ -16,18 +19,78 @@ public boolean isSuccessful() {
     return issues.stream().noneMatch(issue -> issue.getSeverity() == IssueSeverity.Error);
   }
 
-  public ValidationResult addError(String message, Node subject) {
+  public ValidationResult addError(
+      @Nonnull String message, @Nullable ClassifierInstance<?> subject) {
+    Objects.requireNonNull(message, "message should not be null");
     issues.add(new Issue(IssueSeverity.Error, message, subject));
     return this;
   }
 
-  public <S extends Node> ValidationResult checkForError(boolean check, String message, S subject) {
+  public ValidationResult addError(@Nonnull String message, @Nullable String subject) {
+    Objects.requireNonNull(message, "message should not be null");
+    issues.add(new Issue(IssueSeverity.Error, message, subject));
+    return this;
+  }
+
+  public ValidationResult addError(@Nonnull String message) {
+    return addError(message, (String) null);
+  }
+
+  /**
+   * Checks the specified condition and, if the condition evaluates to true, adds an error to the
+   * current validation result with the provided message and subject.
+   *
+   * @param check a boolean condition that, when true, will trigger an error to be added
+   * @param message a non-null string describing the error; must not be null
+   * @param subject an optional subject associated with the error, which can be null.
+   * @return the current ValidationResult, potentially updated with a new error if the condition was
+   *     true
+   * @throws NullPointerException if the message is null
+   */
+  public <S extends ClassifierInstance<?>> ValidationResult checkForError(
```

Comment:
I think I'd like `addErrorIf` more as a name for this method than `checkForError`.

Have you considered passing a “thunk” for the message rather than a `String` that always gets build, regardless of whether `check` is `true` or not.

(Same remarks apply to the other `checkForError` method.)